cpp/src/strings/copying/copying.cu

```diff
@@ -42,6 +44,38 @@ std::unique_ptr<cudf::column> copy_slice(strings_column_view const& strings,
   if (end < 0 || end > strings_count) end = strings_count;
   CUDF_EXPECTS(((start >= 0) && (start < end)), "Invalid start parameter value.");
   strings_count = cudf::util::round_up_safe<size_type>((end - start), step);
+  if (start == 0 && strings.offset() == 0) {
```

Comment:
This does not account for the `step` or the `end` parameters.

Reply:
Nice catch!  This is fixed.